Visual change for case management

jingcheng16 opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
- Make "Edit Form" a secondary button
<img width="636" alt="image" src="https://github.com/user-attachments/assets/0f5b255c-b873-4f96-8250-8b184d5072a3" />

- Make "Add Property" a primary button
- Change "Add Property" button to "Add Case Property" in Case Management tab, and to "Add User Property" in User Management Tab
<img width="953" alt="image" src="https://github.com/user-attachments/assets/7f5e7c99-0075-47b6-b95f-54cc22cd1b02" />

<img width="628" alt="image" src="https://github.com/user-attachments/assets/7a34f237-7068-490d-9fba-c3dc634fc274" />

- Change "Case Property" in User Management Tab to "User Property"
<img width="928" alt="image" src="https://github.com/user-attachments/assets/8638f6bd-c656-4be8-bda9-bddbd1d25b5f" />

In advanced form, say Case Property
<img width="553" alt="image" src="https://github.com/user-attachments/assets/c2f1baa0-9a5b-4dd4-92f0-9423f6f63f50" />


## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Ticket: https://dimagi.atlassian.net/browse/SAAS-16782 and https://dimagi.atlassian.net/browse/SAAS-16848

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Just some button style changes and text change. Tested locally.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No QA


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

jingcheng16 committed:
Add type to differentiate Case Property and User Property

jingcheng16 committed:
Display different header for different properties

jingcheng16 committed:
Display different button for different properties

jingcheng16 committed:
Make add property button primary button

jingcheng16 committed:
Make Edit Form button secondary button

jingcheng16 committed:
prettify

orangejenny reviewed (commented):
This code looks good, but I think you also need to update `advanced/case_config_ui.js` so that this works for advanced forms, too. Check out https://github.com/dimagi/commcare-hq/pull/35733/ and then subsequent change https://github.com/dimagi/commcare-hq/pull/35842/ that fixes advanced forms. (Also, as a bonus you could probably remove the `title` attribute that https://github.com/dimagi/commcare-hq/pull/35733/ added, and instead use the `type` you added here to determine the text updated in that PR.)

jingcheng16 committed:
Always use "Case Property" in advanced forms

Advanced form don't have data.type, so if there is no data.type, we can say it is for case property

jingcheng16 committed:
Remove title since we have type

jingcheng16 commented:
@orangejenny Thank you for catching this! I assume Advanced module can only add case property? Since in #35842  you make the title `self.title = gettext("Save Questions to Case Properties");` 
Fixed advanced form in 1910f8e6836ed48146c3a5dfd9fed9c9ecafa89f and remove title in 4d4699f46c0b830d6017e6a3c665f7eddd47508b
